clients/venice-admin-tool/src/main/java/com/linkedin/venice/AdminTool.java

```diff
@@ -2836,4 +2845,14 @@ private static void printErr(String errorMessage, Map<String, String> customMess
       System.out.println("{\"" + ERROR + "\":\"" + e.getMessage() + "\"}");
     }
   }
+
+  private static PubSubConsumerAdapter getConsumer(Properties consumerProps) {
+    KafkaPubSubMessageDeserializer kafkaPubSubMessageDeserializer = new KafkaPubSubMessageDeserializer(
```

Comment:
Ditto for hear.  Also I see this is static, can we just make this a utility function in common for both this and change capture?

Reply:
As we want to control Xinfra Clients for different component independently, so I think we probably need to have different utility function in different components.

Reply:
We could maybe keep in common if we have getConsumer parameterized by the consumer type the individual component wants?

Something like getConsumer(Properties consumerProperties, String className) (or enum) or just put that as a new variable in consumerProperties?

I'm just not stoked on having the code duplicated even if temporary.

Reply:
This might be a bit difficult here, as in open source venice, we should not have enum with assumption of different pub sub system types. As for different component client or application, it would be better to let the client or application itself to choose its pub sub system type to use. For change capture consumer, we might need to have to add `PubSubConsumerFactory` to the interface of Factory.

Reply:
sounds good.  Lets add that TODO